### DISCUSSION TITLE
Creating a circle ROI through info coming from ExternalizedProperties

### DISCUSSION THREAD

**Discussion by anamargaridampinto** [bonsai-rx/bonsai#1589] [Q&A]
Hi everyone!

I have a behavioral task where I need alternate between 18 circular ROIs, one per trial, which will determine a trigger for a close loop stim. 

I've built already a working bonsai as well as a python script that are inputing several variables as ExternalizedProperties.

So going back on my question, I was thinking the best way to implement this would be to input, for every ROI, the X,Y values of the center of the circle, + a diameter/radius value. 

The CropPolygon function works great, but having 35 externalized variables for every ROI would really be a pain. 

Is there any way to do it like I was imagining or any other that would minimize the nr of variables I would need to input?

Thanks a lot in advance :)

**Comment by chiyu1203** [comment]
As you wrote on another question. I think using csv reader to read through the position of each ROI every time is the most straightforward way to implement this idea.

**Comment by bruno-f-cruz** [comment]
Cant you just use some math to calculate the points of a circle? I would probably use a csharp transform, but if you want to do everything in bonsai, give this a try:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/edab83b2-1f3f-43a7-a15a-1babdd49adaa)

Hope it helps!

[DrawCircle.zip](https://github.com/bonsai-rx/bonsai/files/13228318/DrawCircle.zip)

**Reply by anamargaridampinto** [reply]
I was going to go for the first option you mentioned, but will give this full bonsai one a try. Thanks a lot Bruno, I appreciate it!
